hw/ip/otbn/dv/model/otbn_model.cc

```diff
@@ -15,9 +15,16 @@
 #include "iss_wrapper.h"
 #include "sv_scoped.h"
 
-extern "C" int simutil_get_mem(int index, svBitVecVal *val);
-extern "C" int simutil_set_mem(int index, const svBitVecVal *val);
-extern "C" int otbn_rf_peek(int index, svBitVecVal *val);
+extern "C" {
+// DPI imports, implemented in SystemVerilog
+int simutil_get_mem(int index, svBitVecVal *val);
+int simutil_set_mem(int index, const svBitVecVal *val);
+int otbn_rf_peek(int index, svBitVecVal *val) __attribute__((weak));
+}
+
+// A function pointer to otbn_rf_peek(). Can be NULL if otbn_rf_peek() is not
+// implemented, i.e. if no strong symbol is found at runtime.
+int (*otbn_rf_peek_ptr)(int index, svBitVecVal *val) = &otbn_rf_peek;
```

Comment:
Do you need to do that? Have you tried changing the assertion below to just `assert(otbn_rf_peek)` ?

Reply:
I thought it didn't work when I tried it, but it seems to work now. Not sure what was going on (or if I now missed something), but it at least seems to compile :smile:

Reply:
> it at least seems to compile

Ship it!